sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -225,14 +225,26 @@ abstract class BaseRelation {
    *  java.lang.String -> UTF8String
    *  java.lang.Decimal -> Decimal
    *
-   * If `needConversion` is `false`, buildScan() should return an [[RDD]] of [[InternalRow]]
    *
    * Note: The internal representation is not stable across releases and thus data sources outside
    * of Spark SQL should leave this as true.
    *
-   * @since 1.4.0
+   * @since 1.5.0
+   */
+  def inputNeedConversion: Boolean = true
+
+  /**
+   * Whether does it need to convert scan results to [[Row]] instead of [[InternalRow]].
```

Comment:
`Indicates if the results of a scan need to be converted from a [[Row]] to an [[InternalRow]]`